### PR TITLE
feat: add Lark to repository

### PR DESCRIPTION
**Pull request by first-storm**

### CODE REVIEW

**Comment by first-storm** [comment]
This pull request introduces a new Flatpak package for the Lark application, including the necessary metadata and configuration files. The most important changes include adding a desktop entry, metadata information, and a YAML manifest file for the package.

Metadata and configuration:

* [`manifests/com.larksuite.Lark/com.larksuite.Lark.desktop`](diffhunk://#diff-58b83ac5ff896805c08aa387af321ac86f0705f4345ae18069d43ed75fc4a4d8R1-R21): Added a desktop entry for the Lark application, including details such as name, comment, execution command, and categories.
* [`manifests/com.larksuite.Lark/com.larksuite.Lark.metainfo.xml`](diffhunk://#diff-5037b682469fbb543581acd62b0262ff9573e0874ed97cfe24453a29a77cc4b2R1-R23): Added metadata information for the Lark application, including the developer's name, description, homepage URL, and recommended controls.
* [`manifests/com.larksuite.Lark/com.larksuite.Lark.yaml`](diffhunk://#diff-8267e8d778adadc2077b9781d7489d05c559d659db52e5b80affd7901e41e397R1-R63): Added a YAML manifest file for the Lark Flatpak package, specifying the app ID, runtime, SDK, build commands, and sources.

By the way, are you still maintaining this? Maybe we can work together.

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  - --talk-name=org.gnome.ScreenSaver
+  # Share files
+  - --filesystem=home
+separate-locales: false
+modules:
+  - name: lark
+    buildsystem: simple
+    build-commands:
+      - install -D -m 755 -t /app/bin/ apply_extra lark
+      - install -D -t /app/lib/ /usr/lib/x86_64-linux-gnu/{libbfd-*.so,libsframe.so.1}
```
Comment (pugaizai) [inline manifests/com.larksuite.Lark/com.larksuite.Lark.yaml:26]: use bsdtar instead

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        commands:
+          - ar x Lark-linux.deb
+          - tar -xf data.tar.xz
+          - mv opt/bytedance/lark .
+          - sed '/^exec -a/d' lark/bytedance-lark > lark/bytedance-lark.sh
+          - rm -rf opt usr *.tar.* debian-binary *.deb
+        dest-filename: apply_extra
+      - type: extra-data
+        filename: Lark-linux.deb
+        url: https://sf16-va.larksuitecdn.com/obj/lark-artifact-storage/1d51a0a0/Lark-linux_x64-7.28.10.deb
```
Comment (pugaizai) [inline manifests/com.larksuite.Lark/com.larksuite.Lark.yaml:48]: aarch64 build

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
+app-id: com.larksuite.Lark
+runtime: org.freedesktop.Platform
+runtime-version: '24.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '24.08'
+command: lark
+finish-args:
```
Comment (pugaizai) [inline manifests/com.larksuite.Lark/com.larksuite.Lark.yaml:8]: tray icon?

**Review by pugaizai — COMMENTED** [review]
also add application id to .github/workflows/update.yml build.yml

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml (hunk 2 above)
Comment (pugaizai) [reply manifests/com.larksuite.Lark/com.larksuite.Lark.yaml:48]: 我这里都是尽量要有arm版的

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml (hunk 3 above)
Comment (first-storm) [reply manifests/com.larksuite.Lark/com.larksuite.Lark.yaml:8]: https://docs.flatpak.org/en/latest/desktop-integration.html

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  - --share=network
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  # For camera
+  - --device=all
+  - --talk-name=org.freedesktop.secrets
+  - --talk-name=org.gnome.keyring
+  - --talk-name=org.gnome.ScreenSaver
```
Comment (pugaizai) [inline manifests/com.larksuite.Lark/com.larksuite.Lark.yaml:17]:
```suggestion
  - --talk-name=org.gnome.ScreenSaver
  - --talk-name=org.kde.StatusNotifierWatcher
  - --own-name=org.kde.*
```
你没发现这个软件有托盘的嘛？

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  # Share files
+  - --filesystem=home
+separate-locales: false
+modules:
+  - name: lark
+    buildsystem: simple
+    build-commands:
+      - install -D -m 755 -t /app/bin/ apply_extra lark
+      - install -D -t /app/lib/ /usr/lib/x86_64-linux-gnu/{libbfd-*.so,libsframe.so.1}
+      - install -D -t /app/bin/ /usr/bin/ar
```
Comment (pugaizai) [inline manifests/com.larksuite.Lark/com.larksuite.Lark.yaml]:
```suggestion
```

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      - install -D -t /app/share/icons/hicolor/scalable/apps/ com.larksuite.Lark.svg
+      - install -D -t /app/share/applications/ com.larksuite.Lark.desktop
+      - install -D -t /app/share/metainfo/ com.larksuite.Lark.metainfo.xml
+    only-arches:
+      - x86_64
+      - aarch64
```
Comment (pugaizai) [inline manifests/com.larksuite.Lark/com.larksuite.Lark.yaml]:
```suggestion
```

**Review by pugaizai — CHANGES_REQUESTED** [review]

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        size: 380671576
+        only-arches:
+          - x86_64
+        x-checker-data:
+          type: json
+          url: https://www.larksuite.com/api/downloads
+          version-query: .versions.Linux_deb_x64.version_number
+          url-query: .versions.Linux_deb_x64.download_link
+      - type: extra-data
+        filename: Lark-linux-arm64.deb
```
Comment (pugaizai) [inline manifests/com.larksuite.Lark/com.larksuite.Lark.yaml]:
```suggestion
        filename: Lark-linux.deb
```

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  # For camera
+  - --device=all
+  - --talk-name=org.freedesktop.secrets
+  - --talk-name=org.gnome.keyring
+  - --talk-name=org.gnome.ScreenSaver
+  # Share files
+  - --filesystem=home
```
Comment (pugaizai) [inline manifests/com.larksuite.Lark/com.larksuite.Lark.yaml]: 这个权限希望修改一下，因为会在home文件夹拉💩

**Review by pugaizai — CHANGES_REQUESTED** [review]

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml (hunk 4 above)
Comment (pugaizai) [reply manifests/com.larksuite.Lark/com.larksuite.Lark.yaml:17]: 可以使用`flatpak run --log-session-bus <app-id>`来找找看有没有漏的

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml (hunk 5 above)
Comment (first-storm) [reply manifests/com.larksuite.Lark/com.larksuite.Lark.yaml]: 为啥删掉这个

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml (hunk 8 above)
Comment (first-storm) [reply manifests/com.larksuite.Lark/com.larksuite.Lark.yaml]: 这没办法，因为这是文档软件

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml (hunk 5 above)
Comment (pugaizai) [reply manifests/com.larksuite.Lark/com.larksuite.Lark.yaml]: 你都用不上`ar`了还要来这个干嘛

File: manifests/com.larksuite.Lark/com.larksuite.Lark.yaml (hunk 8 above)
Comment (pugaizai) [reply manifests/com.larksuite.Lark/com.larksuite.Lark.yaml]: 虽然但是，这应该是云文档软件

Comment (pugaizai) [reply manifests/com.larksuite.Lark/com.larksuite.Lark.yaml]: 限定在documents和download就好

**Comment by first-storm** [comment]
I can't get this on GNOME. There is no errors while running with `flatpak run --log-session-bus <app-id>`.

**Comment by pugaizai** [comment]
```
C3: -> org.freedesktop.DBus call org.freedesktop.DBus.NameHasOwner at /org/freedesktop/DBus
Filtering message due to arg0 org.kde.StatusNotifierWatcher, policy: 0 (required 1)
B-1: <- org.freedesktop.DBus return from C3
```

https://docs.flatpak.org/en/latest/desktop-integration.html

chapter StatusNotifier

**Pull request merged**